src/main/java/org/truffleruby/core/string/TBytesKey.java

```diff
@@ -62,4 +92,61 @@ public String toString() {
         return TruffleString.fromByteArrayUncached(bytes, encoding, false).toString();
     }
 
+    private static int hashCode(InternalByteArray byteArray) {
+        return hashCode(byteArray.getArray(), byteArray.getOffset(), byteArray.getLength());
+    }
+
+    // A variant of <code>Arrays.hashCode</code> that allows for selecting a range within the array.
+    private static int hashCode(byte[] bytes, int offset, int length) {
+        if (bytes == null) {
+            return 0;
+        }
```

Comment:
`bytes` can never be null, right? I'll just remove this then.
(it probably comes from Arrays#hashCode checking that)